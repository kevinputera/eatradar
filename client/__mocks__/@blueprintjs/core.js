/**
 * This manual mock only mocks Toaster, Position, and Intent component
 * and is only used in StatusMessage.test.js
 */
const core = jest.requireActual('@blueprintjs/core');

const clearMock = jest.fn();
const showMock = jest.fn();
const Toaster = {
  create: () => ({
    clear: clearMock,
    show: showMock,
  }),
};

module.exports = {
  ...core,
  clearMock,
  showMock,
  Toaster,
};
