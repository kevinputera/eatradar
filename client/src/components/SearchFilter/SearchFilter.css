/* .search-filter {
  display: flex;
  flex-flow: row nowrap;
} */

/* .search-filter-input-wrapper {
  flex-grow: 1;
} */

/* .refresh-button-wrapper {
  margin-left: 5px;
} */
