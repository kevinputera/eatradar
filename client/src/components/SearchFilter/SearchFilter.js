import React from 'react';
import { InputGroup, Button, Tooltip } from '@blueprintjs/core';

import './SearchFilter.css';

function SearchFilter(props) {
  return (
    <div className="search-filter">
      <div className="search-filter-input-wrapper">
        <InputGroup
          large
          leftIcon="search"
          placeholder="Search restaurants"
          value={props.query}
          onChange={props.handleQueryInputChange}
          style={{ borderRadius: '10px' }}
        />
      </div>
      <div className="refresh-button-wrapper">
        <Tooltip position="bottom" content="Update your location">
          <Button
            large
            minimal
            icon="refresh"
            onClick={props.handleRefreshButtonClick}
            style={{ borderRadius: '20px' }}
          />
        </Tooltip>
      </div>
    </div>
  );
}

export default SearchFilter;
