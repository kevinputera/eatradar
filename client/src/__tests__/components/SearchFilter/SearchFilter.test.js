import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { throttle } from 'lodash';
import { InputGroup } from '@blueprintjs/core';

import SearchFilter from '../../../components/SearchFilter/SearchFilter';

jest.mock('lodash');
const throttledFunc = jest.fn();
throttle.mockImplementation(jest.fn(() => throttledFunc));

describe('tests for SearchFilter component', () => {
  beforeEach(() => {
    throttledFunc.mockClear();
  });

  test('render must match snapshot', () => {
    const func = () => {};
    const tree = create(<SearchFilter updateQueryInput={func} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('input change must update input state and InputGroup value prop', () => {
    const event = { target: { value: 'a' } };
    const wrapper = shallow(<SearchFilter />);

    const before = wrapper.find(InputGroup);
    expect(before.prop('value')).toBe('');
    before.simulate('change', event);

    const after = wrapper.find(InputGroup);
    expect(after.prop('value')).toBe('a');
  });

  test('input change must trigger throttled updateQueryInput prop', () => {
    const event = { target: { value: 'a' } };
    const wrapper = shallow(<SearchFilter />);

    const input = wrapper.find(InputGroup);
    expect(throttledFunc).toHaveBeenCalledTimes(0);
    input.simulate('change', event);
    expect(throttledFunc).toHaveBeenCalledTimes(1);
    expect(throttledFunc).toHaveBeenCalledWith('a');
  });

  test('throttle must be created from updateQueryInput prop', () => {
    const func = () => {};
    shallow(<SearchFilter updateQueryInput={func} />);
    expect(throttle).toHaveBeenCalledWith(func, expect.any(Number));
  });
});
