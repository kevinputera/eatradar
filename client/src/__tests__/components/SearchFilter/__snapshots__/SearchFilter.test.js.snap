// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for SearchFilter component render must match snapshot 1`] = `
<div
  className="search-filter"
>
  <div
    className="bp3-input-group bp3-large"
  >
    <span
      className="bp3-icon bp3-icon-search"
      icon="search"
    >
      <svg
        data-icon="search"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          search
        </desc>
        <path
          d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
    <input
      className="bp3-input"
      onChange={[Function]}
      placeholder="Search restaurants"
      style={
        Object {
          "borderRadius": "10px",
          "paddingRight": 10,
        }
      }
      type="search"
      value=""
    />
  </div>
</div>
`;
